pkg/storage/fs/owncloud/owncloud.go

```diff
@@ -473,6 +509,42 @@ func (fs *ocfs) getOwner(internal string) string {
 	return ""
 }
 
+func (fs *ocfs) getUser(ctx context.Context, usernameOrID string) (id *userpb.User, err error) {
+	u := user.ContextMustGetUser(ctx)
+	// check if username matches and id is set
+	if u.Username == usernameOrID && u.Id != nil && u.Id.OpaqueId != "" {
+		return u, nil
+	}
+	// check if userid matches and username is set
+	if u.Id != nil && u.Id.OpaqueId == usernameOrID && u.Username != "" {
+		return u, nil
+	}
+	// look up at the userprovider
+
+	// parts[0] contains the username or userid. use  user service to look up id
+	c, err := pool.GetUserProviderServiceClient(fs.c.UserProviderEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	res, err := c.GetUser(ctx, &userpb.GetUserRequest{
+		UserId: &userpb.UserId{OpaqueId: usernameOrID},
```

Comment:
When configuring the userprovider with ldap you need to configure a filter that checks both: the username attribute as well as the userid attribute.